spec/fixtures/hiera/usergroups.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+nebula::usergroup::membership:
+  examplegroup1:
+  - exampleuser1
+  - exampleuser2
+  examplegroup2:
+  - exampleuser2
+  - exampleuser3
+
+nebula::profile::groups::all_groups:
```

Comment:
What about something like:

```yaml
nebula::groups:
  examplegroup1:
    gid: 1234
    members:
    - exampleuser1
    - exampleuser2
  examplegroup2:
    gid: 2486
    members:
    - exampleuser2
    - exampleuser3
```
That would keep all the information about groups a little more closely together at the expense of making using the data in the profiles/functions/etc slightly more complicated.

Reply:
I went back and forth on this and eventually chose separate hashes because:

1.  It makes the profiles/functions/etc slightly less complicated (particularly in cases like sudo, where we don't want puppet to manage the group itself, but we do want it to manage membership—as it stands, we can just leave it out of the gid group, and we don't need to write any puppet code checking for a gid key-value in each group).
2.  The two lists have different purposes from our perspective. One should only be messed with when we create a new group, but the other is subject to change all the time.

But yeah, disadvantages include the complexity of having two parameters instead of one and a repetition of every group name. I'm not convinced that either is the "right" way to do it. Really, if anything, the right way is to figure out how to handle users and groups outside of hieradata, so I guess I'm inclined to want to think about that instead of this and to stick with the easy solution in the meantime.

Reply:
OK, sounds good.